pdata/pprofile/attributes.go

```diff
@@ -26,3 +30,35 @@ func FromAttributeIndices(table AttributeTableSlice, record attributable) pcommo
 
 	return m
 }
+
+// AddAttribute updates an AttributeTable and a record's AttributeIndices to
+// add a new attribute.
+// The record can by any struct that implements an `AttributeIndices` method.
+func AddAttribute(table AttributeTableSlice, record attributable, key string, value any) error {
+	for i := 0; i < table.Len(); i++ {
+		a := table.At(i)
+
+		if a.Key() == key && value == a.Value().AsRaw() {
+			if i >= math.MaxInt32 {
+				return fmt.Errorf("Attribute %s=%#v has too high an indice to be added to AttributeIndices", key, value)
+			}
+
+			record.AttributeIndices().Append(int32(i)) //nolint:gosec // overflow checked
```

Comment:
Can we avoid identical entries by looping over the AttributeIndices and checking whether `i` already exists?

Reply:
That kind of goes with the concern regarding loops and deletion of attributes I brought up in the PR description.

Reply:
I'd like to hear @bogdandrutu's feedback first.

Reply:
Just to understand the background correctly: when processors add attributes, are these explicitly requested by the user via configuration? If the user is in control and aware, I'd agree that a check for duplication here is not needed.

In any case, we should document the behavior.